src/routes/Organizations/Repository/Tabs/Tags/Filter.tsx

```diff
@@ -10,91 +10,165 @@ import {
   SelectOption,
   SelectVariant,
   Pagination,
+  Modal,
+  ModalVariant,
+  Button,
+  Label,
 } from '@patternfly/react-core';
 import {useState} from 'react';
-import {useRecoilState} from 'recoil';
-import {filterState, paginationState} from 'src/atoms/TagListState';
+import {useRecoilState, useRecoilValue} from 'recoil';
+import {
+  filterState,
+  paginationState,
+  selectedTagsState,
+} from 'src/atoms/TagListState';
+import {deleteTag} from 'src/resources/TagResource';
 
 export function TagsToolbar(props: ToolBarProps) {
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
   const [filter, setFilter] = useRecoilState(filterState);
   const [pagination, setPagination] = useRecoilState(paginationState);
   const [filterOpen, setFilterOpen] = useState(false);
   const [selected, setSelected] = useState([]);
+  const [selectedTags, setSelectedTags] = useRecoilState(selectedTagsState);
 
   const filterTags = (value: string) => {
     setFilter(value);
     setPagination({page: 1, perPage: 25});
   };
 
-  const kebabItems = [<DropdownItem key="delete">Delete</DropdownItem>];
+  const deleteTags = async () => {
+    await Promise.all(
+      selectedTags.map(async (tag) =>
+        deleteTag(props.organization, props.repository, tag),
+      ),
+    );
+    props.loadTags();
+    setIsModalOpen(!isModalOpen);
+    setSelectedTags([]);
+  };
+
+  const kebabItems = [
+    <DropdownItem
+      key="delete"
+      onClick={() => {
+        setIsModalOpen(!isModalOpen);
+      }}
+    >
+      Delete
+    </DropdownItem>,
+  ];
   const [isKebabOpen, setKebabOpen] = useState(false);
 
   return (
-    <Toolbar>
-      <ToolbarContent>
-        <ToolbarItem spacer={{default: 'spacerNone'}}>
-          {/* Need to hook up the functionality of this select */}
-          <Select
-            variant={SelectVariant.checkbox}
-            aria-label="Select Filter Input"
-            onToggle={() => setFilterOpen(!filterOpen)}
-            onSelect={() => {
-              console.log('selected');
+    <>
+      <Modal
```

Comment:
Same with modal,  can you move it into it's own component. It would simplify this file

Reply:
same here